packages/astro/src/vite-plugin-astro/compile.ts

```diff
@@ -37,8 +37,8 @@ export async function compileAstro({
 		// Compile all TypeScript to JavaScript.
 		// Also, catches invalid JS/TS in the compiled output before returning.
 		esbuildResult = await transformWithEsbuild(transformResult.code, compileProps.filename, {
+			...compileProps.viteConfig.esbuild,
 			loader: 'ts',
-			target: 'esnext',
```

Comment:
Since the default target in esbuild is esnext, the explicit specification has been removed.
https://esbuild.github.io/api/#target:~:text=The%20default%20target%20is%20esnext